### PR TITLE
Feature/push/add push record

### PR DESCRIPTION
**Pull request by TalhaAli00**

### CODE REVIEW

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/data/MessageRepository.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package com.walletconnect.push.wallet.data
+
+import com.walletconnect.push.common.storage.data.dao.MessageQueries
+
+class MessageRepository(private val messageQueries: MessageQueries) {
```
Comment (jakubuid) [inline push/sdk/src/main/kotlin/com/walletconnect/push/wallet/data/MessageRepository.kt:5]: Should be internal

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/data/MessageRepository.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        icon: String?,
+        url: String?,
+    ) {
+        messageQueries.insertMessage(requestId, topic, publishedAt, title, body, icon, url)
+    }
+
+    fun getMessagesByTopic(topic: String): List<MessageRecord> =
+        messageQueries.getMessagesByTopic(topic, ::mapToMessageRecord).executeAsList()
+
+    fun deleteMessage(requestId: Long) {
```
Comment (jakubuid) [inline push/sdk/src/main/kotlin/com/walletconnect/push/wallet/data/MessageRepository.kt:22]: All the methods in this calls get to be internal

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/PushWalletEngine.kt [hunk 3]
```diff
@@ -155,6 +151,14 @@ internal class PushWalletEngine(
         )
     }
 
+    fun deleteMessage(requestId: Long, onFailure: (Throwable) -> Unit) {
```
Comment (jakubuid) [inline push/sdk/src/main/kotlin/com/walletconnect/push/wallet/engine/PushWalletEngine.kt:154]: Let's also make all of the engine methods internal

**Review by jakubuid — APPROVED** [review]

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/data/MessageRepository.kt (hunk 2 above)
Comment (TalhaAli00) [reply push/sdk/src/main/kotlin/com/walletconnect/push/wallet/data/MessageRepository.kt:22]: with the file being internal, we don't have to have to each function be internal

File: push/sdk/src/main/kotlin/com/walletconnect/push/wallet/data/MessageRepository.kt (hunk 1 above)
Comment (TalhaAli00) [reply push/sdk/src/main/kotlin/com/walletconnect/push/wallet/data/MessageRepository.kt:5]: done

**Pull request merged**